### PR TITLE
[POC][Tests-Only] Add first visual regression tests

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add first visual regression test for the header of files page.

## Related Issue

- Related https://github.com/owncloud/web/issues/4404

## Motivation and Context

Visual regression testing is a very useful to reduce visual regressions. our e2e tests use dom elements to run the tests but they do not check the UI visually. So even if the colour or tests of the elements are changed the tests will not catch those (more in https://github.com/owncloud/web/issues/4404#issuecomment-736506646). visual regression testing will help reduce these kinds of bugs

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/stepDefinitions/visualContext.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Then('the top bar should match the default baseline', async function() {
+  const backend = client.globals.ocis ? 'ocis' : 'oc10'
+  const element = visualElements[backend].topBar
+  await client.assert.screenshotIdenticalToBaseline(
+    element.selector,
+    element.path,
+    { threshold: 0 },
+    'Matched the top bar of files page'
+  )
```
Comment (Talank) [inline tests/acceptance/stepDefinitions/visualContext.js:27]: cool :1st_place_medal:

**Comment by ownclouders** [comment]
</p></details>

</p></details>

**Comment by ownclouders** [comment]
</p></details>

</p></details>

**Comment by ownclouders** [comment]
</p></details>

**Comment by ownclouders** [comment]
</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISVisual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12694/

<p>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Visual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12694/

<p>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISVisual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12696/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Visual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12697/

<p>


Diff Image: </br>

Actual Image: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISVisual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12697/

<p>


Diff Image: </br>

Actual Image: </br>


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>OCISVisual</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/12703/

<p>


Diff Image: </br>

Actual Image: </br>

Comparing Against: </br>


</p></details>

**Comment by LukasHirt** [comment]
My original thought was that we'll add such visual checks into already existing steps instead of introducing custom scenarios for them. Not to have to write them down in such a way + save us having to spin up another pipeline for it. Also, this could then check in different states of UI without duplicating some steps e.g. check when files list is loaded, check when the sidebar is opened, check when a resource is shared and people accordion is opened... Does that make sense? I did the same approach with unit tests where I'm checking for snapshots of the code - https://github.com/owncloud/web/blob/85e53fc71c4dd32e2830de1349d6d1e1cd1a610a/tests/unit/specs/TopBar.spec.js#L32

**Comment by dpakach** [comment]
@LukasHirt Yes, that approach will work but I see a few potential problems, 
1. Since we will have the checks within existing steps, which are called from many scenarios, we will be doing the same checks from many different places (we can probably solve this by combination having a config option and check the screenshots and cucumber tags).

2. Since we will have the visual regression testing mixed up with other scenarios, we will have a hard time keeping tabs on how much visual regression testing we have, Having them in a separate scenario will give us a proper idea of exactly how much visual regression tests we have.

**Review by individual-it — CHANGES_REQUESTED** [review]
code looks good to me, please document the usage (or create an issue to write docs) and squash commits

File: docs/testing.md [hunk 2]
```diff
@@ -100,6 +100,27 @@ see [available settings](#available-settings-to-be-set-by-environment-variables)
 5. Run `yarn run acceptance-tests-ocis <feature-files-to-test>`.
    The feature files are located in the "tests/acceptance/features" subdirectories.
 
+### Visual Regression Testing
+The test suite consists of snapshots of UI components which can be compared for visual regression testing when running the acceptance tests. These comparisions are done in the existing scenarios. You can check the existing snapshots of the components in the directory `/tests/vrt/baseline`.
+
+#### Running the visual regression tests
+When you run the accpetance tests normally, all the visual regression comparisions are skipped. To run the acceptance test suite with the Visual comparision enabled you need to set the env variable, `VISUAL_TEST` to `true`
```
Comment (individual-it) [inline docs/testing.md:107]:
```suggestion
When you run the acceptance tests as usual, all the visual regression comparisons are skipped. To run the acceptance test suite with the visual comparison enabled you need to set the env variable, `VISUAL_TEST` to `true`
```

**Review by individual-it — CHANGES_REQUESTED** [review]
small typo only

**Review by individual-it — APPROVED** [review]

**Pull request merged**